nada_dsl/nada_types/function.py

```diff
@@ -138,20 +119,7 @@ def store_in_ast(self, ty):
         )
 
 
-def contained_types(ty):
-    """Utility function that calculates the child type for a function argument."""
-
-    origin_ty = getattr(ty, "__origin__", ty)
-    if not issubclass(origin_ty, ScalarType):
-        inner_ty = getattr(ty, "__args__", None)
-        inner_ty = contained_types(inner_ty[0]) if inner_ty else T
-        return origin_ty.init_as_template_type(inner_ty)
-    if origin_ty.mode == Mode.CONSTANT:
-        return origin_ty(value=0)
-    return origin_ty(child=None)
-
-
-def nada_fn(fn, args_ty=None, return_ty=None) -> NadaFunction[T, R]:
+def create_nada_fn(fn, args_ty) -> NadaFunction[T, R]:
```

Comment:
This doesn't stop me from using this as a decorator, `@create_nada_fn`.

Reply:
Is there a way to prevent that?